src/uwtools/resources/rocoto.jsonschema

```diff
@@ -1,44 +1,55 @@
 {
-    "attrs": {
-        "realtime": {
+    "workflow": {
+        "attrs": {
+            "realtime": {
+                "type": "string",
+                "enum": [
+                    "true",
+                    "false",
+                    "T",
+                    "F",
+                    "t",
+                    "f",
+                    "True",
+                    "False",
+                    "FALSE",
+                    "TRUE"
+                ]
+            }
+            "scheduler": {
+                "type": "string",
+                "enum": [
+                    "sge",
+                    "lsf",
+                    "lsfcray",
+                    "ll",
+                    "torque",
+                    "moabtorque",
+                    "moab",
+                    "pbspro",
+                    "slurm",
+                    "cobalt"
+                ]
+            }
+            "cycylethrottle": {
+                "type": "integer",
+                "minimum": 0
+            }
+            "taskthrottle": {
+                "type": "integer",
+                "minimum": 0
+            }
+        }
+        "cycledefs": {
             "type": "string",
-            "enum": [
-                "true",
-                "false",
-                "T",
-                "F",
-                "t",
-                "f",
-                "True",
-                "False",
-                "FALSE",
-                "TRUE"
-            ]
+            "pattern": "^\[\d{10},\d{10},(0[0-9]|1[0-9]|2[0-4]):([0-5][0-9]):([0-5][0-9])\]$"
```

Comment:
What do we envision appearing in UW YAML that would be validated by `cycledefs` here? In the Rocoto XML I see
```
<cycledef group="at_start">201907010000 201907010000 24:00:00</cycledef>
<cycledef group="forecast">201907010000 201907010000 24:00:00</cycledef>
```
So, one or more (per the RelaxNG schema) of these `<cycledef>` entities, each with an optional `group` attribute, with two `yyyymmddHHMM` items and then a `HH:MM:SS` item. How would that appear in UW YAML?

Also, does JSON Schema support defining patterns at a higher, reusable level? We've got `yyyymmddHHMM` repeated here, which already suggests that we should give it a name; and looking at the full realized XML I'm seeing a `HH:MM:SS` pattern repeated as the value of the `<wallclock>` entity under `<task>`, so we'll need that, too. I see that RelaxNG supports defining reusable patterns, and that's used in the Rocoto XML schema. Can we do that with JSON Schema, too?

Reply:
Here, the YAML spec might look like this:
```
  cycledefs:
    groupname:
      - !startstopfreq ['2022102000', ‘2023102018’, ‘06:00:00’]
    groupname2:
      - !startstopfreq ['2022102000', ‘2023102018’, ‘24:00:00’]
      - !startstopfreq ['2022102006', ‘2023102018’, ‘24:00:00’]
```

So, it would need to support an object where keys were arbitrary and values are lists of these date strings.

Reply:
Circling back on checking if JSON Schema supports defining patterns and reusing them:

It seems that JSONSchema supports, however it may not be _easily_. I'm finding [discussion](https://github.com/json-schema-org/json-schema-spec/issues/886) online that others want to do a similar thing here, where we can add our regex expression in a constant and reuse the variable when needed. 

There is an option to use `$ref` which after much googling will allow us to

Reply:
I'd like to see what you come up with. Anything that cuts down on repetition of schema elements that should always be the same thing would be a plus.